src/detectors/builtin/sendInLoop.ts

```diff
@@ -0,0 +1,400 @@
+    statement: AstStatement,
+    warnings: MistiTactWarning[],
+    inLoop: boolean,
+    analyzedFunctions: Set<string>,
+  ): void {
+    const condition = statement as AstCondition;
+    for (const stmt of condition.trueStatements) {
+      this.analyzeStatement(stmt, warnings, inLoop, analyzedFunctions);
+    }
+    if (condition.falseStatements) {
+      for (const stmt of condition.falseStatements) {
+        this.analyzeStatement(stmt, warnings, inLoop, analyzedFunctions);
+      }
+    }
+    if (condition.elseif) {
+      this.analyzeStatement(
+        condition.elseif,
+        warnings,
+        inLoop,
+        analyzedFunctions,
+      );
+    }
+  }
+
+  /**
+   * Handles try-catch statements by analyzing their try and catch blocks.
+   */
+  private handleTryCatchStatement(
+    statement: AstStatement,
+    warnings: MistiTactWarning[],
+    inLoop: boolean,
+    analyzedFunctions: Set<string>,
+  ): void {
+    if (statement.kind === "statement_try") {
+      const tryStmt = statement as AstStatementTry;
+      for (const stmt of tryStmt.statements) {
+        this.analyzeStatement(stmt, warnings, inLoop, analyzedFunctions);
+      }
+    } else if (statement.kind === "statement_try_catch") {
+      const tryCatchStmt = statement as AstStatementTryCatch;
+      for (const stmt of tryCatchStmt.statements) {
+        this.analyzeStatement(stmt, warnings, inLoop, analyzedFunctions);
+      }
+      for (const stmt of tryCatchStmt.catchStatements) {
+        this.analyzeStatement(stmt, warnings, inLoop, analyzedFunctions);
+      }
+    }
+  }
+
+  /**
+   * Handles expression statements by checking for send function calls if inside a loop.
+   */
+  private handleExpressionStatement(
+    statement: AstStatementExpression,
+    warnings: MistiTactWarning[],
+    inLoop: boolean,
+    analyzedFunctions: Set<string>,
+  ): void {
+    if (inLoop) {
+      this.checkForSendFunctions(
+        statement.expression,
+        warnings,
+        analyzedFunctions,
+      );
+    }
+  }
+
+  /**
+   * Checks if an expression contains a send function call and records a warning.
+   */
+  private checkForSendFunctions(
+    expr: AstExpression,
+    warnings: MistiTactWarning[],
+    analyzedFunctions: Set<string>,
+  ): void {
+    const functionName = this.extractFunctionName(expr);
+    if (functionName) {
+      if (this.isSendFunction(functionName)) {
+        warnings.push(
+          this.makeWarning(
+            `Send function '${functionName}' called inside a loop`,
+            expr.loc,
+            {
+              suggestion:
+                "Consider refactoring to avoid calling send functions inside loops.",
+            },
+          ),
+        );
+      } else if (this.functionDefinitions.has(functionName)) {
+        if (!analyzedFunctions.has(functionName)) {
+          analyzedFunctions.add(functionName);
+          const func = this.functionDefinitions.get(functionName)!;
+          if ("statements" in func) {
```

Comment:
Just don't add asm functions in `this.functionDefinitions` and we won't need this condition.

Reply:
removed.